libr/core/cmd_type.c

```diff
@@ -1209,24 +1209,42 @@ static int cmd_type(void *data, const char *input) {
 		}
 		break;
 	case 't': {
-		if (!input[1]) {
-			char *name = NULL;
-			SdbKv *kv;
-			SdbListIter *iter;
-			SdbList *l = sdb_foreach_list (TDB, true);
-			ls_foreach (l, iter, kv) {
-				if (!strcmp (sdbkv_value (kv), "typedef")) {
-					if (!name || strcmp (sdbkv_value (kv), name)) {
-						free (name);
-						name = strdup (sdbkv_key (kv));
+		if (!input[1] || input[1]=='j') {
+        		bool json = false;
+    			if (input[1]=='j') {
+				r_cons_print ("{");
+            		}
+            		char *name = NULL;
+            		SdbKv *kv;
+            		SdbListIter *iter;
+            		SdbList *l = sdb_foreach_list (TDB, true);
+            		ls_foreach (l, iter, kv) {
+            		if (!strcmp (sdbkv_value (kv), "typedef")) {
+                		if (!name || strcmp (sdbkv_value (kv), name)) {
+                    			free (name);
+                    			name = strdup (sdbkv_key (kv));
+                    			if (!input[1]) {
 						r_cons_println (name);
 					}
-				}
-			}
-			free (name);
-			ls_free (l);
-			break;
-		}
+                    		else {
+                        		if (json) {
+                            			r_cons_print(",");
+                        		}                                                        
+                    			const char *q = sdb_fmt ("typedef.%s", name);
+                        		const char *res = sdb_const_get (TDB, q, 0);
+                        		r_cons_printf("\"%s\":\"%s\"", name, res);                                                        
```

Comment:
missing space before (

Reply:
we may use the json api that will be introduced after release. otherwise we will be suffering from escaping chars and encoding issues here and there.